### PR TITLE
Added SLE15-SP5 targets, newer Rubocop

### PR DESCRIPTION
**Pull request by lslezak**
- Originally I just wanted to fix failing GitHub Actions caused by using an obsolete Ruby version (Factory moved from 2.7 to 3.1 quite some time ago)
- But the old Rubocop is not compatible with Ruby 3.1 so we need to upgrade to a newer one...
- Also some data in the *.gemspec were obsolete (email, URL)...
- And when touching this we could also update the OBS targets...

### CODE REVIEW

File: libyui-rake.gemspec [hunk 1]
```diff
@@ -17,31 +19,36 @@
...
+
   # dependencies
-  spec.add_runtime_dependency("rake", "> 10.0", "< 99")
-  spec.add_runtime_dependency("packaging_rake_tasks", "~> 1.5")
   # "~> 1.5"  means ">= 1.5 and < 2" (thanks to darix)
+  spec.add_runtime_dependency("packaging_rake_tasks", "~> 1.5")
+  spec.add_runtime_dependency("rake", "> 10.0", "< 99")
+  # enable MFA (2FA) authentication at rubygems.org
+  # see https://guides.rubygems.org/mfa-requirement-opt-in/
+  spec.metadata["rubygems_mfa_required"] = "true"
```
Comment (lslezak) [inline libyui-rake.gemspec:53]: This was automatically added by Rubocop, but IMHO it makes sense. See https://guides.rubygems.org/mfa-requirement-opt-in/ for more details.

Personally I have enabled MFA. @jreidinger @mvidner @imobachgs, what about you? Is this change OK?

Comment (jreidinger) [reply libyui-rake.gemspec:53]: well, I would have to enable it, but not critical for me.

Comment (imobachgs) [reply libyui-rake.gemspec:53]: It is OK for me.

**Review by imobachgs — APPROVED** [review]

File: libyui-rake.gemspec (hunk 1 above)
Comment (mvidner) [reply libyui-rake.gemspec:53]: Well, I haven't tried this option yet. Can we lock ourselves out if the other maintainers haven't either? :rofl:

Comment (lslezak) [reply libyui-rake.gemspec:53]: You cannot lock out yourselves, if you do not use MFA then just enable it and then you can do whatever you need.

Comment (jreidinger) [reply libyui-rake.gemspec:53]: fine for me

**Review by jreidinger — APPROVED** [review]

**Pull request merged**